plugins/catalog-backend/src/service/DefaultEntitiesCatalog.ts

```diff
@@ -111,24 +111,21 @@ function addCondition(
   filter: EntitiesSearchFilter,
   negate: boolean = false,
   entityIdField = 'entity_id',
-) {
+): void {
+  const key = filter.key.toLowerCase();
+  const values = filter.values?.map(v => v.toLowerCase());
+
   // NOTE(freben): This used to be a set of OUTER JOIN, which may seem to
   // make a lot of sense. However, it had abysmal performance on sqlite
   // when datasets grew large, so we're using IN instead.
   const matchQuery = db<DbSearchRow>('search')
     .select('search.entity_id')
-    .where({ key: filter.key.toLowerCase() })
+    .where({ key })
     .andWhere(function keyFilter() {
-      if (filter.values) {
-        if (filter.values.length === 1) {
-          this.where({ value: filter.values[0].toLowerCase() });
-        } else {
-          this.andWhere(
-            'value',
-            'in',
-            filter.values.map(v => v.toLowerCase()),
-          );
-        }
+      if (values?.length === 1) {
+        this.where({ value: values.at(0) });
```

Comment:
Any reason for going for `.at`, otherwise let's keep it
```suggestion
        this.where({ value: values[0] });
```

Reply:
The reason why I chose `.at(x)` is because its safer to use compared to directly accessing an array. `.at(x)` returns `T | undefined`; whereas directly accessing returns `T` or an array access out of bounds error. In this case it doesn't matter because we check the array length beforehand; however, it still is a safer method to use.